### PR TITLE
Update haskell layer documenation.

### PR DESCRIPTION
**Pull request by maxigit**
Improve haskell layer documentation

- add syntax checking section
- improve FAQ to solve common syntaxk checking problems.

### CODE REVIEW

File: layers/+lang/haskell/README.org [hunk 1]
```diff
@@ -342,6 +359,47 @@ type is inserted. On a symbol "foo" without definition,
...
+This is the standard spacemacs way to do syntax checking and the most elaborate. flyckeck displays errors, warning, info, including symbols on the left or right side of the buffer
+and tooltips if possible.
+Errors are also available in the error window (~SPC e l~) and you can navigate between errors using ~SPC e n~ and ~SPC e p~.
+Flycheck can be toggle on/off with ~SPC t s~.
+Flycheck has different haskell checker : haskell-ghc, haskell-stackghc and haskell-hlint.
+It normally detect automatically the good one, but if it doesn't work, you can change it using =flycheck-select-checker= (~SPC e s~).
+
+** HLint
+HLint is a linter for Haskell. It doesn't detect error (as long as it can parse the file) but detect bad coding style and code smell.
+HLint is done
```
Comment (d12frosted) [inline layers/+lang/haskell/README.org]: unfinished?

File: layers/+lang/haskell/README.org [hunk 2]
```diff
@@ -342,6 +359,47 @@ type is inserted. On a symbol "foo" without definition,
...
+  - hlint (via flycheck)
+  - ghc-mod
+  - haskell-mode interactive
+    
+As all those components can be active at the same time, it can be tricky to know which component is display which message,
+especially when they disagree or one is not working. Only flycheck errors (ghc and hlint) are displayed in the error list and
+can be navigated using the standard spacemacs key bindings even though errors from other mode might highlight error in the buffer.
+
+** Flycheck   
+This is the standard spacemacs way to do syntax checking and the most elaborate. flyckeck displays errors, warning, info, including symbols on the left or right side of the buffer
```
Comment (d12frosted) [inline layers/+lang/haskell/README.org]: is it worth describing what `flycheck` is here? https://github.com/syl20bnr/spacemacs/tree/develop/layers/%2Bcheckers/syntax-checking is a good pointing

though it's good to tell people that they have 3 different checkers

File: layers/+lang/haskell/README.org [hunk 3]
```diff
@@ -342,6 +359,47 @@ type is inserted. On a symbol "foo" without definition,
...
+Errors are also available in the error window (~SPC e l~) and you can navigate between errors using ~SPC e n~ and ~SPC e p~.
+Flycheck can be toggle on/off with ~SPC t s~.
+Flycheck has different haskell checker : haskell-ghc, haskell-stackghc and haskell-hlint.
+It normally detect automatically the good one, but if it doesn't work, you can change it using =flycheck-select-checker= (~SPC e s~).
+
+** HLint
+HLint is a linter for Haskell. It doesn't detect error (as long as it can parse the file) but detect bad coding style and code smell.
+HLint is done
+
+** GHC-MOD
```
Comment (bgamari) [inline layers/+lang/haskell/README.org]: `GHC-MOD` shouldn't be capitalized.

File: layers/+lang/haskell/README.org [hunk 4]
```diff
@@ -408,3 +465,62 @@ This is intended behavior in =haskell-indentation-mode=. If you want to reset in
 (advice-add 'haskell-indentation-newline-and-indent
             :after 'haskell-indentation-advice)
 #+END_SRC
+** Flycheck displays HLInt warnings but not errors
+HLint checker is called *after* normal flycheck checker even if the checkers fails. Check the [[Flycheck doesn't work]] section.
+** I can see highlighted errors but they don't appear in the errors list
+The errors list is only set by flycheck. You are probably seeing highlightings coming from either ghc-mode or haskell-mode.
+Check the [[Flycheck doesn't work]] section.
+** Flycheck doesn't work
+You can check what is wrong with flychec with the =flycheck-compile= command.
```
Comment (d12frosted) [inline layers/+lang/haskell/README.org]: > flychec

flycheck :)

File: layers/+lang/haskell/README.org (hunk 2 above)
Comment (maxigit) [inline layers/+lang/haskell/README.org]: I don't think it hurts. Some people might comes to spacemacs from haskell and don't know anything about flycheck.
I'm not sure a link to the github doc is thing either. You probably can submit a PR to correct it once this one is accepted.

File: layers/+lang/haskell/README.org (hunk 3 above)
Comment (maxigit) [inline layers/+lang/haskell/README.org]: OK.

File: layers/+lang/haskell/README.org (hunk 4 above)
Comment (maxigit) [inline layers/+lang/haskell/README.org]: done.

File: layers/+lang/haskell/README.org (hunk 2 above)
Comment (d12frosted) [inline layers/+lang/haskell/README.org]: I mean, why duplicate info, when we can link to existing Spacemacs documentation (`syntax-checking` layer)? If you think that that layer documentation is missing something - just improve it. My point is that `flycheck` is not `haskell` only thing, it's used all over the places in Spacemacs. 

P. S. don't get me wrong - I am all for better description of Flycheck (in my opinion, it should have a section in general documentation file). Describing haskell-related parts of flycheck should be in haskell layer, obviously, just don't get to general here :)

**Comment by d12frosted** [comment]
And as general, please refer to [Readme Conventions](https://github.com/syl20bnr/spacemacs/blob/master/doc/CONVENTIONS.org#spacing-in-documentation). I know that haskell layer violates spacing in some places - better to fix them than create new violations :)

**Comment by d12frosted** [comment]
Just noticed that you closed this PR :-P

**Pull request closed without merging**